### PR TITLE
updates to fonts

### LINKED ISSUE #8472: inline noto-subset font load failed in Firefox
## Observed behavior

In Firefox on Mac, only the full and common fonts appear to be loading. The base64-encoded css-inlined subset font is not.

Probably related to https://github.com/khaledhosny/ots/issues/52

## Errors and logs

Normal weight:

```
downloadable font: glyf: zero-length table (font-family: "noto-subset" style:normal weight:400 stretch:100 src index:0) source: data:application/x-font-woff;charset=utf-8;base64,AAEAAAARAQAABAAQR0RFRgBjAAQAABRAAAAAJkdQT1NEdkx1AAAUaAAAACBHU1VCkxWCFgAAFIgAAA … FAABAAAAAAABAAAACgAcAB4AAURGTFQACAAEAAAAAP//AAAAAAAAAAEAAAAKADIANAAEREZMVAAaY3lybAAkZ3JlawAkbGF0bgAkAAQAAAAA//8AAAAAAAAAAAAAAAA=
downloadable font: rejected by sanitizer (font-family: "noto-subset" style:normal weight:400 stretch:100 src index:0) source: data:application/x-font-woff;charset=utf-8;base64,AAEAAAARAQAABAAQR0RFRgBjAAQAABRAAAAAJkdQT1NEdkx1AAAUaAAAACBHU1VCkxWCFgAAFIgAAA … FAABAAAAAAABAAAACgAcAB4AAURGTFQACAAEAAAAAP//AAAAAAAAAAEAAAAKADIANAAEREZMVAAaY3lybAAkZ3JlawAkbGF0bgAkAAQAAAAA//8AAAAAAAAAAAAAAAA=
downloadable font: font load failed (font-family: "noto-subset" style:normal weight:400 stretch:100 src index:0) source: data:application/x-font-woff;charset=utf-8;base64,AAEAAAARAQAABAAQR0RFRgBjAAQAABRAAAAAJkdQT1NEdkx1AAAUaAAAACBHU1VCkxWCFgAAFIgAAA … FAABAAAAAAABAAAACgAcAB4AAURGTFQACAAEAAAAAP//AAAAAAAAAAEAAAAKADIANAAEREZMVAAaY3lybAAkZ3JlawAkbGF0bgAkAAQAAAAA//8AAAAAAAAAAAAAAAA=
```

Bold weight:

```
downloadable font: glyf: zero-length table (font-family: "noto-subset" style:normal weight:700 stretch:100 src index:0) source: data:application/x-font-woff;charset=utf-8;base64,AAEAAAARAQAABAAQR0RFRgBjAAQAABQoAAAAJkdQT1NEdkx1AAAUUAAAACBHU1VCkxWCFgAAFHAAAA … FAABAAAAAAABAAAACgAcAB4AAURGTFQACAAEAAAAAP//AAAAAAAAAAEAAAAKADIANAAEREZMVAAaY3lybAAkZ3JlawAkbGF0bgAkAAQAAAAA//8AAAAAAAAAAAAAAAA=
downloadable font: rejected by sanitizer (font-family: "noto-subset" style:normal weight:700 stretch:100 src index:0) source: data:application/x-font-woff;charset=utf-8;base64,AAEAAAARAQAABAAQR0RFRgBjAAQAABQoAAAAJkdQT1NEdkx1AAAUUAAAACBHU1VCkxWCFgAAFHAAAA … FAABAAAAAAABAAAACgAcAB4AAURGTFQACAAEAAAAAP//AAAAAAAAAAEAAAAKADIANAAEREZMVAAaY3lybAAkZ3JlawAkbGF0bgAkAAQAAAAA//8AAAAAAAAAAAAAAAA=
downloadable font: font load failed (font-family: "noto-subset" style:normal weight:700 stretch:100 src index:0) source: data:application/x-font-woff;charset=utf-8;base64,AAEAAAARAQAABAAQR0RFRgBjAAQAABQoAAAAJkdQT1NEdkx1AAAUUAAAACBHU1VCkxWCFgAAFHAAAA … FAABAAAAAAABAAAACgAcAB4AAURGTFQACAAEAAAAAP//AAAAAAAAAAEAAAAKADIANAAEREZMVAAaY3lybAAkZ3JlawAkbGF0bgAkAAQAAAAA//8AAAAAAAAAAAAAAAA=
```

## Expected behavior

The inline font is intended to prevent a flash of unstyled text and improve render performance.

## User-facing consequences

Possibly slower loads or shifting typography

## Steps to reproduce

Load Kolibri on Firefox on Mac

## Context

latest develop

### PR DESCRIPTION
**Pull request by indirectlylit**
## Summary

- Upgrading the `FontTools` dependency
- Simplify font-loading code for "modern" browsers
- Ensure that there are no conflicting glyphs in generated subsets
- Updated to latest Noto fonts. Adds many new scripts:
  - Adlam
  - Balinese
  - Bamum
  - Gondi
  - Kayah Li
  - Lisu
  - Medefaidrin
  - Meetei Mayek
  - Ol Chiki
  - Sora Sompeng
  - Sundanese
  - Tai Tham

## References

* fixes https://github.com/learningequality/kolibri/issues/8472

## Reviewer guidance

This update should be an overall improvement of font handling, but I haven't thoroughly checked impact on a range of browsers. If someone could help check the status of these issues in particular I'd appreciate it:

- https://github.com/learningequality/kolibri/issues/4515
- ~https://github.com/learningequality/kolibri/issues/5939~ fixed in #8768
- https://github.com/learningequality/kolibri/issues/7799

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by indirectlylit** [comment]
@rtibbles I'll take a look at https://github.com/learningequality/kolibri/issues/7805 as a follow-up

**Comment by indirectlylit** [comment]
converting to draft – found an issue

**Comment by indirectlylit** [comment]
ready for review

**Comment by radinamatic** [comment]
Tested the EXE asset in IE11, Firefox & Chrome on Windows 7, and Edge, Firefox & Chrome on Windows 10; found only 2 obvious issues:

1. IE11/Windows 7: labels for Urdu and Hindi are missing in the language selection modal ONLY when the UI is in English.
    And IE11 still seems to be loading all full non-Latin font families... 
   
   ie11

2. Chrome on Windows 10, font color in modal has been changed...? Firefox and Edge seem unaffected. 
    
    
    Win10 (start)  Running  - Oracle VM VirtualBox_006

**Comment by radinamatic** [comment]
Hm, in this latest build I can confirm that non-Latin fonts are not loading by default on IE11, but there is still the issue of the Urdu and Hindi labels missing in the change language modal... 😕 

ie11-2

**Comment by indirectlylit** [comment]
thanks for testing! will investigate early next week

**Comment by indirectlylit** [comment]
> IE11/Windows 7: labels for Urdu and Hindi are missing in the language selection modal ONLY when the UI is in English.
And IE11 still seems to be loading all full non-Latin font families...

This should be addressed in c5c240008374a5f392ead26a5107424845d545cc

> Chrome on Windows 10, font color in modal has been changed...? Firefox and Edge seem unaffected.

I'm not reproducing this in Browserstack.  Any additional guidance?


| modal | browser details |
|--|--|
| image | image |

File: kolibri/core/assets/src/utils/setupAndLoadFonts.js [hunk 1]
```diff
@@ -91,10 +85,11 @@ function addFontStylesheetLink(href) {
 }
 
 export default function setupAndLoadFonts() {
-  if (passesRequirements(browser, modernFontBrowsers)) {
+  if (passesRequirements(browser, fontSwapSupportingBrowsers)) {
     addFontStylesheetLink(plugin_data.fullCSSFileModern);
+    loadFullFontsImmediately();
```
Comment (rtibbles) [inline kolibri/core/assets/src/utils/setupAndLoadFonts.js:90]: Could return early here and avoid the `else` clause.

**Review by rtibbles — COMMENTED** [review]
No concerns from the code side - once this passes manual QA we're good to merge.

**Comment by radinamatic** [comment]
With the latest build I'm not able to replicate past issues any more! 👏🏽 

Labels for Urdu and Hindi in the language selection ✔️  | Font color in modal on Chrome ✔️ 
--- | ---
Win7 (start)  Running  - Oracle VM VirtualBox_011 | I retested in 2 completely different VMs just to be sure <br><br> Win10 (start)  Running  - Oracle VM VirtualBox_007

**Review by radinamatic — APPROVED** [review]
:shipit: 💯

File: kolibri/core/assets/src/utils/setupAndLoadFonts.js (hunk 1 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/utils/setupAndLoadFonts.js:90]: yes! sounds like a comment I might usually leave :)

**Pull request merged**